### PR TITLE
Make types in Microsoft.AspNetCore.Mvc.ViewFeatures internal

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test.csproj [hunk 1]
```diff
@@ -8,4 +8,8 @@
     <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.Views.TestCommon\Microsoft.AspNetCore.Mvc.Views.TestCommon.csproj" />
     <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.TestDiagnosticListener\Microsoft.AspNetCore.Mvc.TestDiagnosticListener.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="ViewFeatures\" />
+  </ItemGroup>
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test.csproj]: Suggest undoing this addition.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Buffers/PagedBufferedTextWriter.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Buffers
 {
     public class PagedBufferedTextWriter : TextWriter
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Buffers/PagedBufferedTextWriter.cs]: Could `IViewBufferScope.CreateWriter(...)` return a `TextWriter`? That should allow this type to be `internal`.

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.ViewFeatures/Buffers/PagedBufferedTextWriter.cs]: Yeah, that seems like a reasonable design. Added it as a note to the API review issue

**Comment by pranavkm** [comment]
`TempDataMvcOptionsSetup` was accidental. Added the rest to the list

**Pull request merged**